lib/statsd/instrument.rb

```diff
@@ -62,26 +62,26 @@ def self.generate_metric_name(metric_name, callee, *args)
       metric_name.respond_to?(:call) ? metric_name.call(callee, args).gsub('::', '.') : metric_name.gsub('::', '.')
     end
 
-    if Process.respond_to?(:clock_gettime)
-      # @private
-      def self.current_timestamp
-        Process.clock_gettime(Process::CLOCK_MONOTONIC)
-      end
-    else
-      # @private
-      def self.current_timestamp
-        Time.now
-      end
+    # Even though this method is considered private, and is no longer used internally,
+    # applications in the wild rely on it. As a result, we cannot remove this method
+    # until the next major version.
+    #
+    # @deprecated Use Process.clock_gettime(Process::CLOCK_MONOTONIC) instead.
+    def self.current_timestamp
```

Comment:
@Bringer128 do you think we need to keep this method around because people depend on it in their apps? Or can we :fire: it?

Reply:
I removed this previously and we needed to revert as people were relying on it; as the comment says we should bump the major version number if we want to remove this.